pallets/block-rewards-hybrid/src/tests.rs

```diff
@@ -0,0 +1,462 @@
+            collators_percent: Perbill::zero(),
+            adjustable_percent: Perbill::one(),
+            ideal_dapps_staking_tvl: Perbill::from_percent(50), // this is irrelevant
+        };
+        assert!(reward_config.is_consistent());
+        assert_ok!(BlockReward::set_configuration(
+            RuntimeOrigin::root(),
+            reward_config.clone()
+        ));
+
+        for _block in 1..=10 {
+            let init_balance_state = FreeBalanceSnapshot::new();
+            let rewards = Rewards::calculate(&reward_config);
+
+            BlockReward::on_timestamp_set(0);
+
+            let final_balance_state = FreeBalanceSnapshot::new();
+            init_balance_state.assert_distribution(&final_balance_state, &rewards);
+        }
+    })
+}
+
+/// Represents free balance snapshot at a specific point in time
+#[derive(PartialEq, Eq, Clone, RuntimeDebug)]
+struct FreeBalanceSnapshot {
+    treasury: Balance,
+    collators: Balance,
+    stakers: Balance,
+    dapps: Balance,
+}
+
+impl FreeBalanceSnapshot {
+    /// Creates a new free balance snapshot using current balance state.
+    ///
+    /// Future balance changes won't be reflected in this instance.
+    fn new() -> Self {
+        Self {
+            treasury: <TestRuntime as Config>::Currency::free_balance(
+                &TREASURY_POT.into_account_truncating(),
+            ),
+            collators: <TestRuntime as Config>::Currency::free_balance(
+                &COLLATOR_POT.into_account_truncating(),
+            ),
+            stakers: <TestRuntime as Config>::Currency::free_balance(
+                &STAKERS_POT.into_account_truncating(),
+            ),
+            dapps: <TestRuntime as Config>::Currency::free_balance(
+                &DAPPS_POT.into_account_truncating(),
+            ),
+        }
+    }
+
+    /// `true` if all free balances equal `Zero`, `false` otherwise
+    fn is_zero(&self) -> bool {
+        self.treasury.is_zero()
+            && self.collators.is_zero()
+            && self.stakers.is_zero()
+            && self.dapps.is_zero()
+    }
+
+    /// Asserts that `post_reward_state` is as expected.
+    ///
+    /// Increase in balances, based on `rewards` values, is verified.
+    ///
+    fn assert_distribution(&self, post_reward_state: &Self, rewards: &Rewards) {
+        assert_eq!(
+            self.treasury + rewards.treasury_reward,
+            post_reward_state.treasury
+        );
+        assert_eq!(
+            self.stakers + rewards.base_staker_reward + rewards.adjustable_staker_reward,
+            post_reward_state.stakers
+        );
+        assert_eq!(
+            self.collators + rewards.collators_reward,
+            post_reward_state.collators
+        );
+        assert_eq!(self.dapps + rewards.dapps_reward, post_reward_state.dapps);
+    }
+}
+
+/// Represents reward distribution balances for a single distribution.
+#[derive(PartialEq, Eq, Clone, RuntimeDebug)]
+struct Rewards {
```

Comment:
One thing I'd suggest is to make the _real_ `Rewards` struct public for tests, and write dedicated tests for it. Then you can use it normally in tests.

Not a change request, only a suggestion.